src/main/java/com/google/research/bleth/simulator/AbstractSimulation.java

```diff
@@ -51,6 +52,11 @@ public void run() {
         writeSimulationStats();
     }
 
+    /** Returns the simulation's Id. */
+    public String getId() {
+        return id;
+    }
```

Comment:
Why should a `Simulation` object return its `simultaionId`? Is it for testing purposes?

Reply:
Otherwise we won't be able to read the simulation later - we should store this id somewhere.

Reply:
Why not?
When writing the `SimulationMetadata` its key property is actually the simulationId. Whenever we want to retrieve a simulation Id it can be retrieved from the `SimulationMetadata` entity.

For example, when visualizing a simulation we first retrieve all `SimulationMetadata` entities from the db. Then the user chooses a simulation and we can retrieve the simulation's Id (which is the entity's key) and roundsNum in order to iteratively fetch the `ReadBoardState` servlet.

Reply:
I'll leave it for now and we'll decide later.